crates/polars-core/src/series/arithmetic/list_borrowed.rs

```diff
@@ -455,59 +310,56 @@ mod inner {
                     self.swapped = false;
                     self._finish_impl_dispatch()
                 },
-                (BinaryOpApplyType::PrimitiveToList, Broadcast::Right) => {
-                    // We materialize the list columns with `new_from_index`, as otherwise we'd have to
-                    // implement logic that broadcasts the offsets and validities across multiple levels
-                    // of nesting. But we will re-use the materialized memory to store the result.
+                (BinaryOpApplyType::ListToList, Broadcast::Left) => {
```

Comment:
drive-by - changed the order of this enum matching to the same order of the comment (expand the diff above to see the comment)